packages/core/src/browser/shell/application-shell.ts

```diff
@@ -561,9 +561,16 @@ export class ApplicationShell extends Widget {
      * to widgets; these need to be transformed before the layout can be serialized.
      */
     getLayoutData(): ApplicationShell.LayoutData {
+        const pinned: boolean[] = [];
+
+        toArray(this.mainPanel.widgets()).forEach((a, i) => {
+            pinned[i] = a.title.className.indexOf('theia-mod-pinned') >= 0;
+        });
```

Comment:
Is the goal only to restore the pinned state for the main-area? I think we should aim to restore the pinned state for the entire layout as the current behavior is confusing for users.

Also, do you mind extracting this functionality to a method, it will make it easier for downstream apps or extensions to customize without needing to re-implement `getLayoutData` entirely.

Reply:
Will extract that. Side panels get saved&restored as well.

Reply:
I'd think all panels, for example the bottom panel looks a lot like the `main` area and it would be confusing for users to pin widgets there and not see them restored.

Reply:
Ah, forgot about that one, sorry!